packages/foundry/contracts/oracles/crab/CrabOracle.sol

```diff
@@ -0,0 +1,142 @@
+// SPDX-License-Identifier: BUSL-1.1
+pragma solidity >=0.8.13;
+
+import "@yield-protocol/utils-v2/contracts/access/AccessControl.sol";
+import "@yield-protocol/utils-v2/contracts/cast/CastBytes32Bytes6.sol";
+import "@yield-protocol/utils-v2/contracts/token/IERC20Metadata.sol";
+import "../../interfaces/IOracle.sol";
+
+interface ICrabStrategy {
+    function totalSupply() external view returns (uint256);
+
+    /**
+     * @notice get the vault composition of the strategy
+     * @return operator
+     * @return nft collateral id
+     * @return collateral amount
+     * @return short amount
+     */
+    function getVaultDetails()
+        external
+        view
+        returns (
+            address,
+            uint256,
+            uint256,
+            uint256
+        );
+}
+
+/**
+ * @title CrabOracle
+ * @notice Oracle to fetch Crab-ETH exchange amounts
+ */
+contract CrabOracle is IOracle, AccessControl {
+    using CastBytes32Bytes6 for bytes32;
+    ICrabStrategy crabStrategy;
+    IOracle uniswapV3Oracle;
+    bytes6 public constant weth = 0x303000000000;
+    bytes6 public crab;
+    bytes6 public oSQTH;
+
+    event SourceSet(
+        bytes6 crab_,
+        bytes6 oSQTH_,
+        ICrabStrategy crabStrategy_,
+        IOracle uniswapV3Oracle_
+    );
+
+    /**
+     * @notice Set crabstrategy & uniswap source
+     */
+    function setSource(
+        bytes6 crab_,
+        bytes6 oSQTH_,
+        ICrabStrategy crabStrategy_,
+        IOracle uniswapV3Oracle_
+    ) external auth {
+        crab = crab_;
+        oSQTH = oSQTH_;
+        crabStrategy = crabStrategy_;
+        uniswapV3Oracle = uniswapV3Oracle_;
+        emit SourceSet(crab_, oSQTH_, crabStrategy_, uniswapV3Oracle_);
+    }
+
+    /**
+     * @notice Retrieve the value of the amount at the latest oracle price.
+     * Only `crabId` and `ethId` are accepted as asset identifiers.
+     */
+    function peek(
+        bytes32 base,
+        bytes32 quote,
+        uint256 baseAmount
+    )
+        external
+        view
```

Comment:
Simply adding `view` is the only differentiation between `get` and `peek`?

Reply:
In `IOracle` get didn't have `view` as in some of the oracles there is a storage change expected.